src/Microsoft.DotNet.Build.Tasks/Targets/versioning.targets

```diff
@@ -92,13 +94,16 @@
       File="$(AssemblyInfoFile)"
       Lines="@(AssemblyInfoUsings);@(AssemblyInfoLines)"
       Overwrite="true" />
-
-    <ItemGroup>
-      <Compile Include="$(AssemblyInfoFile)" />
-      <FileWrites Include="$(AssemblyInfoFile)" />
-    </ItemGroup>
   </Target>
 
+  <ItemGroup Condition="'$(GenerateAssemblyInfo)'=='true'">
+    <Compile Include="$(AssemblyInfoFile)">
```

Comment:
What happens if you open a project before the _AssemblyInfo.cs file is generated? Does it show up as missing? Does it refresh correctly after the first build?